transformation_test/transformation_test.go

```diff
@@ -238,9 +300,34 @@ func generateFluentBitConfigs(ctx context.Context, name string, transformationTe
 	}
 
 	components = append(components, input)
+	uidComponentsIndex := 0
+
+	// Only one (or empty) `Receiver` can be set in a transformation test.
+	if transformationTest.Receiver != nil {
+		if transformationTest.Receiver.InputLogParser.LoggingProcessor == nil {
+			return nil, fmt.Errorf("input_log_parser is required with test_receiver.")
+		}
+		// Append parser to mock an "input" receiver output.
+		inputLogParser := transformationTest.Receiver.InputLogParser.LoggingProcessor
+		components = append(components, inputLogParser.Components(ctx, flbTag, strconv.Itoa(uidComponentsIndex))...)
+		uidComponentsIndex += 1
 
-	for i, t := range transformationTest {
-		components = append(components, t.Components(ctx, flbTag, strconv.Itoa(i))...)
+		if transformationTest.Receiver.TestReceiver.LoggingReceiver == nil {
+			return nil, fmt.Errorf("test_receiver is required.")
+		}
+		// Append logging processors from receiver "post-processing" step.
+		receiverComponents := transformationTest.Receiver.TestReceiver.LoggingReceiver.Components(ctx, flbTag)
```

Comment:
Manipulating the otel/fluent-bit components after they're already constructed feels like a giant hack. Can't we achieve the same result without any hacks by refactoring the receiver-under-test into a receiver + processor that is only registered in tests, and then using that processor in a transformation test?

Reply:
Thanks for the comment! I like the idea of registering `processor`s only for the purpose of testing. This would avoid any need to update `transformation_test.go`. The current change present here seems to need further discussion in the design. I'll close this PR in favour of : https://github.com/GoogleCloudPlatform/ops-agent/pull/1977 .

We can discuss later if adding a `receiver` section is still needed.